st2api/st2api/controllers/v1/packs.py

```diff
@@ -178,14 +179,42 @@ def post(self, pack_register_request):
 class PackSearchController(object):
 
     def post(self, pack_search_request):
+
+        proxy_config = self._get_proxy_config(pack_search_request)
+
         if hasattr(pack_search_request, 'query'):
             packs = packs_service.search_pack_index(pack_search_request.query,
-                                                    case_sensitive=False)
+                                                    case_sensitive=False,
+                                                    proxy_config=proxy_config)
             return [PackAPI(**pack) for pack in packs]
         else:
-            pack = packs_service.get_pack_from_index(pack_search_request.pack)
+            pack = packs_service.get_pack_from_index(pack_search_request.pack,
+                                                     proxy_config=proxy_config)
             return PackAPI(**pack) if pack else []
 
+    def _get_proxy_config(self, pack_search_request):
+        proxy_config = None
+        if hasattr(pack_search_request, 'proxy_config'):
```

Comment:
I removed the part where I get proxy config from API. I originally wanted to support --proxy --cert from st2 pack CLI but we decided it can get confusing to do that especially if CLI is running on a different server and has to talk via a HTTP proxy to the server. I forgot to remove the API change though. Now I removed it.